src/main/java/org/gooru/insights/services/CSVFileWriterServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package org.gooru.insights.services;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.gooru.insights.constants.APIConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+@Service
+public class CSVFileWriterServiceImpl implements CSVFileWriterService{
+	
+	@Autowired
+	private BaseConnectionService baseConnectionService;
+	
+	private static final Logger logger = LoggerFactory.getLogger(CSVFileWriterServiceImpl.class);
+	
+	public BaseConnectionService getBaseConnectionService() {
+		return baseConnectionService;
+	}
+	
+	@Override
+	public void generateCSVReport(Set<String> headerKeys, List<Map<String, Object>> rowList, String fileAbsolutePath, String delimiter, Boolean isNewFile) throws FileNotFoundException {
+		
+		if(delimiter == null || delimiter.isEmpty()) {
+			delimiter = APIConstants.PIPE;
+		}
+		
+		@SuppressWarnings("resource")
+		PrintStream stream = new PrintStream(new BufferedOutputStream(new FileOutputStream(new File(fileAbsolutePath), true)));
```

Comment:
Can we catch the exception and throw here  instead of throwing @ method level  ? And also PrintStream  object need to close on finally block to avoid memory leak.